### DISCUSSION TITLE
Which Frontend?

### DISCUSSION THREAD

**Discussion by glowkeeper** [directus/directus#3867] [Archive]
After having a bit of a "doh!" moment when wondering why my directus install was failing to connect to my non-existent database, I now have directus delivering some MySQL content - which is super-exciting! So much potential!

Anyway - to cut a long story short, on Friday I need to demo a frontend mobile app' where a user logs into directus and displays some data.

Currently, I'm using _react_ as my frontend js framework, so a quick poke around Google seems to suggest Gatsby and its directus plugin. Indeed, I might have already decided on that had https://github.com/directus/gatsby-source resolved  - what's happened to that?

In terms of authenticating, I found this: https://www.npmjs.com/package/@pnodev/nuxt-directus - which suggests nuxt/vue.js might be a good idea - especially since the directus admin' app itself is written in Vue, so there's a great example out there :) Would be cool to give Vue a spin, too.

But what framework/tooling would anyone recommend for fronting up directus? Any recommendations gratefully received!

x

**Comment by rijkvanzanten** [comment]
Heya!

Directus doesn't require any framework or plugin to work properly, so the true answer to your question is: whatever you prefer.

That being said, based on:

> Currently, I'm using react as my frontend js framework

I'd say stick with what you know 🙂 React is more than decent to put together a POC app that pulls from / inserts into Directus. You can use any http-library to interface with Directus (like `fetch` / `XMLHttpRequest` / `axios` / etc), or use the Directus JS SDK if you want something a little more "managed" https://docs.directus.io/reference/sdk-js.html

**Reply by glowkeeper** [reply]
Yeah - you were right - tight time-frame = stick to what you know :) So I have a _rather limited but functional_ React-based frontend up and running that uses `fetch` to access the directus RESTful API, where a user can login and then use the token supplied to display data from a couple of tables. 

So far, I'm only querying a local instance of directus, but my experience is good so far - it seems nice and responsive. Thanks!
